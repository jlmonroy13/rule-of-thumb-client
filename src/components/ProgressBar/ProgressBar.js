import React from 'react';
// import PropTypes from 'prop-types';

const ProgressBar = () => {
  return <div className="progress-bar">ProgressBar</div>;
};

ProgressBar.propTypes = {};

export default ProgressBar;
