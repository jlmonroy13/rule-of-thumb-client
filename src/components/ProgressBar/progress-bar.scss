.progress-bar {
  &__lapsed {
    width: 35%;
    background-color: rgba($black, 0.3);
    display: flex;
    justify-content: flex-end;
    padding-right: 20px;
    align-items: center;
    position: relative;

    &::after {
      position: absolute;
      right: -12px;
      content: '';
      width: 0;
      height: 0;
      border-style: solid;
      border-width: 8px 0 8px 12px;
      border-color: transparent transparent transparent rgba($black, 0.3);
    }
  }

  &__remaining {
    @include font-size(37);

    flex-grow: 1;
    background-color: rgba($white, 0.7);
    line-height: 50px;
    font-weight: 300;
    padding: 0 20px;
  }
}
