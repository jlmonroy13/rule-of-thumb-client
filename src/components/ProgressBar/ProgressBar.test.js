/**
 *
 * Tests for ProgressBar
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import ProgressBar from './index';

const props = {
  label: 'closing in',
  value: 22,
  valueLabel: 'days',
};

describe('<ProgressBar />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<ProgressBar {...props} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<ProgressBar {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
