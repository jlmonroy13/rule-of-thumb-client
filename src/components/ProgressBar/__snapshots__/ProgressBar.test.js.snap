// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> Should render and match the snapshot 1`] = `
<div
  class="d-flex"
>
  <div
    class="progress-bar__lapsed"
  >
    <span
      class="text-white text-uppercase text-small"
    >
      closing in
    </span>
  </div>
  <div
    class="progress-bar__remaining"
  >
    <b
      class="mr-2"
    >
      22
    </b>
    <span>
      days
    </span>
  </div>
</div>
`;
